tldw_Server_API/app/core/Evaluations/recipe_runs_jobs_worker.py

```
@@ -0,0 +1,688 @@
+def _coerce_media_ids(run_config: dict[str, Any], dataset: list[dict[str, Any]]) -> list[int]:
+    explicit_media_ids = run_config.get("media_ids")
+    if isinstance(explicit_media_ids, list) and explicit_media_ids:
+        return [int(media_id) for media_id in explicit_media_ids]
+
+    derived_media_ids: set[int] = set()
+    for sample in dataset:
+        for expected_id in sample.get("expected_ids") or []:
+            try:
+                derived_media_ids.add(int(expected_id))
+            except (TypeError, ValueError):
+                continue
```

Ruff (0.15.7)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)